cms/djangoapps/contentstore/views/item.py

```diff
@@ -702,10 +702,14 @@ def _delete_orphans(course_usage_key, user_id, commit=False):
     """
     store = modulestore()
     items = store.get_orphans(course_usage_key)
+    branch = course_usage_key.branch
     if commit:
         for itemloc in items:
-            # need to delete all versions
-            store.delete_item(itemloc, user_id, revision=ModuleStoreEnum.RevisionOption.all)
+            revision = ModuleStoreEnum.RevisionOption.all
+            # specify branches when deleting orphans
+            if branch == ModuleStoreEnum.BranchName.published:
+                revision = ModuleStoreEnum.RevisionOption.published_only
+            store.delete_item(itemloc, user_id, revision=revision)
```

Comment:
If you don't specify a branch, then it'll work the way it used to work. This just allows you to delete orphans only on the published branch if you specify the branch in the course key.

So
```python
./manage.py cms delete_orphans course-v1:org+course+run
```
will do what it always did, while
```python
./manage.py cms delete_orphans course-v1:org+course+run+branch@published-branch
```
will delete orphans on the published branch